kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -560,6 +574,67 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
             );
         }
 
+        @Override
+        ColumnVector visitLike(final Like like) {
+            PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(like);
+            ColumnVector left = argResults.leftResult;
+            ColumnVector right = argResults.rightResult;
+            int numRows = argResults.rowCount;
+            boolean[] result = new boolean[numRows];
+            boolean[] nullability = new boolean[numRows];
+            for (int rowId = 0; rowId < numRows; rowId++) {
+                String leftOp = left.getString(rowId);
+                String rightOp = right.getString(rowId);
+                if (!Objects.isNull(leftOp) && !Objects.isNull(rightOp)) {
+                    nullability[rowId] = false;
+                    String rightOpRegexPattern =
+                            escapeLikeRegex(rightOp, like.getEscape());
+                    result[rowId] = leftOp.matches(rightOpRegexPattern);
+                } else {
+                    nullability[rowId] = true;
+                    result[rowId] = false;
+                }
+            }
+            return new DefaultBooleanVector(numRows,
+                    Optional.of(nullability), result);
+        }
+
+        private String escapeLikeRegex(String pattern, char escape) {
+            final int len = pattern.length();
+            final StringBuilder javaPattern = new StringBuilder(len + len);
+            for (int i = 0; i < len; i++) {
+                char c = pattern.charAt(i);
+
+                if (c == escape) {
+                    if (i == (pattern.length() - 1)) {
+                        throw invalidEscapeSequence(pattern, i);
+                    }
+                    char nextChar = pattern.charAt(i + 1);
+                    if ((nextChar == '_')
+                            || (nextChar == '%')
+                            || (nextChar == escape)) {
+                        javaPattern.append(
+                                Pattern.quote(Character.toString(nextChar)));
+                        i++;
+                    } else {
+                        throw invalidEscapeSequence(pattern, i);
+                    }
+                } else if (c == '_') {
+                    javaPattern.append('.');
+                } else if (c == '%') {
+                    javaPattern.append(".*");
+                } else {
+                    javaPattern.append(Pattern.quote(Character.toString(c)));
+                }
+
+            }
+            return "(?s)" + javaPattern;
+        }
+
+        private RuntimeException invalidEscapeSequence(String s, int i) {
```

Comment:
Add this to `DefaultEngineErrors`. This is the place containing all utility methods to throw user-facing exceptions.

Reply:
done.